zio-http/src/main/scala/zio/http/ZClient.scala

```diff
@@ -744,6 +756,18 @@ object ZClient {
     }
   }
 
+  final implicit class ClientSyntax(val client: Client) extends AnyVal {
```

Comment:
You could generalize this a bit and just require the In type to be `Body` rather than fixing all the type parameters.

However, it strikes me this will probably work better with tooling if added as a method on `ZClient` that requires implicit evidence of type equality (like `List#flatten`).

Reply:
Oh wait, another thing we could do: why are the original methods which require a body curried? Maybe we could just express them as multiple parameter?

It is not like most people will use curly braces `{ }` to define the body.

Reply:
Not sure why they are curried. I'll change it.

Reply:
Probably the idea was to define functions for different endpoints like `val store = client.post("/user") _` but I think one would either write a wrapper class anyway, or generate it from OpenAPI etc.